optimum/onnxruntime/modeling_decoder.py

```diff
@@ -648,39 +703,440 @@ class ORTModelForCausalLM(ORTModelDecoder, GenerationMixin):
+                    model_id,
+                    [r"^((?!decoder).)*.onnx", pattern],
+                    argument_name=None,
+                    subfolder=subfolder,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                )
+                file_name = decoder_path.name
+
+            MODEL_TO_PATCH_FOR_PAST = {
+                "bloom",
+                "mpt",
+                "llama",
+                "blenderbot-small",
+                "blenderbot",
+                "opt",
+                "pegasus",
+                "bart",
+            }
+            if file_name == ONNX_DECODER_WITH_PAST_NAME and config.model_type in MODEL_TO_PATCH_FOR_PAST:
+                raise ValueError(
+                    f"{ONNX_DECODER_WITH_PAST_NAME} not supported for the following architecture : {', '.join(MODEL_TO_PATCH_FOR_PAST)}. Please re-export your model or set use_cache=False."
+                )
+
+            regular_file_names = []
+            for name in [ONNX_WEIGHTS_NAME, ONNX_DECODER_WITH_PAST_NAME if use_cache else ONNX_DECODER_NAME]:
+                regular_file_names += ORTModelForCausalLM._generate_regular_names_for_filename(name)
+
+            if file_name not in regular_file_names:
+                logger.warning(
+                    f"The ONNX file {file_name} is not a regular name used in optimum.onnxruntime that are {regular_file_names}, the "
+                    f"{cls.__name__} might not behave as expected."
+                )
+
+        if config.model_type == "bloom":
+            init_cls = ORTBloomForCausalLM
+        elif config.model_type == "mpt":
+            init_cls = ORTMPTForCausalLM
+        elif config.model_type == "opt":
+            init_cls = ORTOPTForCausalLM
         else:
-            outputs = self.decoder_with_past(
-                input_ids=input_ids[:, -1:],
-                past_key_values=past_key_values,
-                attention_mask=attention_mask,
-                labels=labels,
-                position_ids=position_ids,
+            init_cls = ORTModelForCausalLM
+
+        model_cache_path, preprocessors = cls._cached_file(
+            model_path=model_path,
+            use_auth_token=use_auth_token,
+            revision=revision,
+            force_download=force_download,
+            cache_dir=cache_dir,
+            file_name=file_name,
+            subfolder=subfolder,
+            local_files_only=local_files_only,
+        )
+        new_model_save_dir = model_cache_path.parent
+
+        # model_save_dir can be provided in kwargs as a TemporaryDirectory instance, in which case we want to keep it
+        # instead of the path only.
+        if model_save_dir is None:
+            model_save_dir = new_model_save_dir
+
+        # Since v1.7.0 decoder with past models have fixed sequence length of 1
+        # To keep these models compatible we set this dimension to dynamic
```

Comment:
I don't understand what happens here. Why is it needed to patch the ONNX models? Why is the output logits sequence length = 1 for with-past ONNX models an issue with the decoder_model.onnx/decoder_with_past_model.onnx?

Reply:
When loading a model previously exported, we will only load the decoder_with_past_model if `use_cache=True` and `use_merged=False`. Since v1.7.0 decoder_with_past_model have a fixed sequence length of 1, to keep these models compatible we set this dimension to dynamic, this is needed for the first generation

Reply:
> When loading a model previously exported, we will only load the decoder_with_past_model

Is the motivation code simplification? How about merged model? Is the inference dropped as well?

> Since v1.7.0 decoder_with_past_model have a fixed sequence length of 1, to keep these models compatible we set this dimension to dynamic, this is needed for the first generation

I would rather raise an error as unsupported (if we stop supporting) and suggesting to re-export the models.